src/Uno.UWP/Devices/Midi/MidiInPort.wasm.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Devices.Enumeration.Internal;
+using Uno.Devices.Midi.Internal;
+using Uno.Foundation;
+
+namespace Windows.Devices.Midi
+{
+	public partial class MidiInPort
+	{
+		private const string JsType = "Windows.Devices.Midi.MidiInPort";
+
+		private readonly string _managedId;
+
+		private readonly static ConcurrentDictionary<string, MidiInPort> _instanceSubscriptions =
+			new ConcurrentDictionary<string, MidiInPort>();
+
+		private MidiInPort(string deviceId, string managedId)
+		{
+			DeviceId = deviceId;
+			_managedId = managedId;
+		}
+
+		partial void StartMessageReceived()
+		{
+			_instanceSubscriptions.TryAdd(_managedId, this);
+			var addListenerCommand = $"{JsType}.startMessageListener('{_managedId}')";
+			WebAssemblyRuntime.InvokeJS(addListenerCommand);
+		}
+
+		partial void StopMessageReceived()
+		{
+			_instanceSubscriptions.TryRemove(_managedId, out _);
+			var removeListenerCommand = $"{JsType}.stopMessageListener('{_managedId}')";
+			WebAssemblyRuntime.InvokeJS(removeListenerCommand);
+		}
+
+		[Preserve]
+		public static int DispatchMessage(string managedId, string serializedMessage, double timestamp)
+		{
+			if (!_instanceSubscriptions.TryGetValue(managedId, out var port))
+			{
+				throw new InvalidOperationException("This instance is not listening to MIDI input.");
+			}
+
+			var managedTimestamp = TimeSpan.FromMilliseconds(timestamp);
+
+			var splitMessage = serializedMessage.Split(':');
+
+			var message = new byte[splitMessage.Length];
+			for (int i = 1; i < splitMessage.Length; i++)
+			{
+				message[i - 1] = byte.Parse(splitMessage[i]);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)